### PR TITLE
overhauled escape_string's doc and added missing mapping

### PR DESCRIPTION
**Pull request by carstenbauer**
See #22286.

### CODE REVIEW

**Comment by crstnbr** [comment]
Something else I should do here, or is it just that no one has found the time to review this PR yet?

**Comment by fredrikekre** [comment]
> Something else I should do here, or is it just that no one has found the time to review this PR yet?

Likely the latter.

File: base/strings/io.jl [hunk 1]
```diff
@@ -251,11 +251,15 @@ escape_nul(s::AbstractString, i::Int) =
...
 """
-    escape_string([io,] str::AbstractString[, esc::AbstractString]) -> AbstractString
+    escape_string(str::AbstractString[, esc::AbstractString]) -> AbstractString
 
 General escaping of traditional C and Unicode escape sequences.
 Any characters in `esc` are also escaped (with a backslash).
-See also [`unescape_string`](@ref).
+The reverse is [`unescape_string`](@ref).
+
+    escape_string(io, str::AbstractString[, esc::AbstractString]) -> Void
```
Comment (fredrikekre) [inline base/strings/io.jl]: Typically we list both signatures at the top together, or split into two separate docstrings.

Comment (crstnbr) [reply base/strings/io.jl]: I prefer the latter. Changed it accordingly.

**Comment by crstnbr** [comment]
Julia freebsd ci error seems to be unrelated.

File: base/strings/io.jl [hunk 2]
```diff
@@ -494,15 +505,4 @@ function convert(::Type{String}, chars::AbstractVector{Char})
...
-                else
-                    write(io, c)
-                    c = d
-                end
-            end
-            write(io, c)
-        end
-    end)
-end
+                d, state = nex
```
Comment (KristofferC) [inline base/strings/io.jl]: What happened here?

**Comment by KristofferC** [comment]
Part of a function is chopped off in the end of the file. Also, it would be nice to have test that checks the different type combinations and the return type.

**Comment by crstnbr** [comment]
I don't know how that happend...

I'll add some tests later today.

File: test/strings/io.jl [hunk 3]
```diff
@@ -251,9 +260,3 @@ using the `tf.gather operation`.  For example:
     "                    end), name, \"NonMaxSuppression\") ",
     "            tf.Tensor(tf.Operation(desc))",
     "        end"]
-
-for i = 1:10
-    buf = IOBuffer()
-    print(buf, join(s22021, "\n"))
-    @test isvalid(String, take!(buf))
-end
```
Comment (crstnbr) [inline test/strings/io.jl]: Again ...

Comment (crstnbr) [reply test/strings/io.jl]: Oh, I think I know why this happened - sync problem while working remotely.

**Comment by crstnbr** [comment]
Tests added.

**Comment by crstnbr** [comment]
IMHO this should be good to go.

**Review by vtjnash — APPROVED** [review]

File: base/strings/io.jl [hunk 4]
```diff
@@ -251,13 +251,21 @@ escape_nul(s::AbstractString, i::Int) =
...
+    escape_string(str::AbstractString[, esc::AbstractString]) -> AbstractString
 
 General escaping of traditional C and Unicode escape sequences.
 Any characters in `esc` are also escaped (with a backslash).
-See also [`unescape_string`](@ref).
+The reverse is [`unescape_string`](@ref).
 """
-function escape_string(io, s::AbstractString, esc::AbstractString)
+escape_string(s::AbstractString, esc::AbstractString) = sprint(endof(s), escape_string, s, esc)
+escape_string(s::AbstractString) = sprint(endof(s), escape_string, s, "\"")
```
Comment (StefanKarpinski) [inline base/strings/io.jl]: This seems to be the wrong default set of `esc` characters:
```jl
julia> sprint(io->escape_string(io, "\""))
"\""

julia> escape_string("\"")
"\\\""
```

**Pull request merged**